### DISCUSSION TITLE
Going crazy over .NET dipendencies

### DISCUSSION THREAD

**Discussion by esseeffeenne** [OpenTabletDriver/OpenTabletDriver#2834] [General]
I have been trying to reinstall OpenTableDriver on my Linux desktop ever since I updated the OS, and .NET problems have been bugging me nonstop.
The Main problem I come down to is:
**1**. the generate_rules script exits telling me I need to add 'net6.0' to the TargetFrameworks of OpenTabletDriver.Tools.udev/OpenTabletDriver.Tools.udev.csproj
**2**. I add 'net6.0' to TargetFrameworks and it then tells me that OpenTabletDriver.Desktop only depends on 'net7.0'. so I remove 'net6.0' and I am back at problem 1.

**PC Info**:
I am on Linux Mint 22 (Vanessa)
dotnet MsBuild version: 17.6.3
dotnet NuGet version: 6.6.0.66
dotnet sdks: 6.0.411, 7.0.305
dotnet runtimes: 6.0.15, 6.0.16, 6.0.19, 7.0.8, both AspNetCore.App and NETCore.App

I run dotnet restore multiple times before running the script
I updated and upgraded dotnet mutiple times

**ERRORS**:
~/Github/OpenTabletDriver$ ./generate-rules.sh 
Generating udev rules...
/home/****/Github/OpenTabletDriver/OpenTabletDriver.Tools.udev/OpenTabletDriver.Tools.udev.csproj : error NU1201: Project OpenTabletDriver.Desktop is not compatible with net6.0 (.NETCoreApp,Version=v6.0). Project OpenTabletDriver.Desktop supports: net7.0 (.NETCoreApp,Version=v7.0)

The build failed. Fix the build errors and run again.
_(with either 'net7.0;net6.0' or 'net7.0' as TargetFrameworks)_

~/Github/OpenTabletDriver$ ./generate-rules.sh 
Generating udev rules...
/usr/share/dotnet/sdk/7.0.305/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(266,5): error NETSDK1005: Assets file '/home/****/Github/OpenTabletDriver/OpenTabletDriver.Tools.udev/obj/project.assets.json' doesn't have a target for 'net6.0'. Ensure that restore has run and that you have included 'net6.0' in the TargetFrameworks for your project. [/home/****/Github/OpenTabletDriver/OpenTabletDriver.Tools.udev/OpenTabletDriver.Tools.udev.csproj::TargetFramework=net6.0]

The build failed. Fix the build errors and run again.
_(with only 'net6.0' as TargetFrameworks)_

**Comment by X9VoiD** [comment]
Which branch or commit are you on? `OpenTabletDriver.Tools.udev` does not use `net6.0` on `master`. If anything, CI never complained about any errors when generating rules so this is on your end.
